src/main/java/gov/acwi/wqp/etl/Application.java

```
@@ -30,61 +32,68 @@
+				|| ExitStatus.FAILED.getExitCode().contentEquals(exitStatus.getExitCode())
+				|| ExitStatus.STOPPED.getExitCode().contentEquals(exitStatus.getExitCode())) {
 			throw new RuntimeException("Job did not complete as planned.");
 		}
 	}
 
-	protected JobParametersBuilder getJobParametersBuilder() {
+	protected JobParameters getJobParametersBuilder(String jobIdDate) {
 		return new JobParametersBuilder(jobExplorer)
-				.addString(DATASOURCE, DATASOURCE_STORET, true);
+				// this constant could have originally been one of the unused args
+				.addString(DATASOURCE, DATASOURCE_STORET, true)
+				// this is the new date based job ID
+				.addString(JOB_ID, jobIdDate, true)
+				.toJobParameters();
 	}
 
-	protected JobExecution reStartJob() throws Exception {
-		JobParameters parameters = getJobParametersBuilder()
-				.addLong(JOB_ID, jobIncrementer.getCurrent(), true)
-				.toJobParameters();
+	protected ExitStatus startJob(String jobIdDate) throws Exception {
+		JobParameters parameters = getJobParametersBuilder(jobIdDate);
+
 		try {
-			return jobLauncher.run(job, parameters);
+			return jobLauncher.run(job, parameters).getExitStatus();
 		} catch (JobInstanceAlreadyCompleteException e) {
-			LOG.info("Job " + job.getName() + " #" + parameters.getString(JOB_ID) + " has already completed successfully.");
-			return startNewJobInstance();
+			// log the new date job ID
+			LOG.info("Job " + job.getName() + " for '" + parameters.getString(JOB_ID) + "' has already completed successfully.");
 		}
-	}
 
-	protected JobExecution startNewJobInstance() throws Exception {
-		JobParameters parameters = getJobParametersBuilder()
-				.getNextJobParameters(job)
-				.toJobParameters();
-		LOG.info("Attempting to start run job " + job.getName() + " #" + parameters.getString(JOB_ID));
-		return jobLauncher.run(job, parameters);
+		return ExitStatus.COMPLETED; // Is this correct? Seems so to me. If it was already done then things are A.O.K.
```

Should this be in the catch block? We only want completed if JobInstanceAlreadyCompleteException, not other  exceptions. (Will we loose the actual exit status if a different exception is thrown?)

---

We could. Any other exception would throw rather than return. The actual job will return its result.

---

Note that I added this return. It was not there prior and I commented it to ensure it is a correct enhancement.